src/core_plugins/region_map/public/__tests__/region_map_visualization.js

```diff
@@ -0,0 +1,391 @@
+    tables: [
+      {
+        columns: [{
+          'aggConfig': {
+            'id': '2',
+            'enabled': true,
+            'type': 'terms',
+            'schema': 'segment',
+            'params': { 'field': 'geo.dest', 'size': 5, 'order': 'desc', 'orderBy': '1' }
+          }, 'title': 'geo.dest: Descending'
+        }, {
+          'aggConfig': { 'id': '1', 'enabled': true, 'type': 'count', 'schema': 'metric', 'params': {} },
+          'title': 'Count'
+        }],
+        rows: [['CN', 26], ['IN', 17], ['US', 6], ['DE', 4], ['BR', 3]]
+      }
+    ]
+  };
+
+  beforeEach(ngMock.module('kibana'));
+  beforeEach(ngMock.inject((Private, $injector) => {
+
+    Vis = Private(visModule.VisProvider);
+    RegionMapsVisualization = Private(RegionMapsVisualizationProvider);
+    indexPattern = Private(LogstashIndexPatternStubProvider);
+
+    ChoroplethLayer.prototype._makeJsonAjaxCall = async function () {
+      //simulate network call
+      return new Promise((resolve)=> {
+        setTimeout(() => {
+          resolve(worldJson);
+        }, 10);
+      });
+    };
+
+    const serviceSettings = $injector.get('serviceSettings');
+    sinon.stub(serviceSettings, '_getManifest', function (url) {
+      let contents = null;
+      if (url.startsWith(tmsManifestUrl)) {
+        contents = tmsManifest;
+      } else if (url.startsWith(vectorManifestUrl)) {
+        contents = vectorManifest;
+      } else if (url.startsWith(manifestUrl)) {
+        contents = manifest;
+      }
+      return {
+        data: contents
+      };
+    });
+
+
+
+  }));
+
+
+  afterEach(function () {
+    ChoroplethLayer.prototype._makeJsonAjaxCall = _makeJsonAjaxCallOld;
+  });
+
+
+  describe('RegionMapVisualization - basics', function () {
+
+    beforeEach(async function () {
+      setupDOM('512px', '512px');
+
+      imageComparator = new ImageComparator();
+
+
+      vis = new Vis(indexPattern, {
+        type: 'region_map'
+      });
+
+      vis.params.selectedJoinField = { 'name': 'iso2', 'description': 'Two letter abbreviation' };
+      vis.params.selectedLayer = {
+        'attribution': '<p><a href="http://www.naturalearthdata.com/about/terms-of-use">Made with NaturalEarth</a> | <a href="https://www.elastic.co/elastic-maps-service">Elastic Maps Service</a></p>&#10;',
+        'name': 'World Countries',
+        'format': 'geojson',
+        'url': 'https://staging-dot-elastic-layer.appspot.com/blob/5715999101812736?elastic_tile_service_tos=agree&my_app_version=7.0.0-alpha1',
+        'fields': [{ 'name': 'iso2', 'description': 'Two letter abbreviation' }, {
+          'name': 'iso3',
+          'description': 'Three letter abbreviation'
+        }, { 'name': 'name', 'description': 'Country name' }],
+        'created_at': '2017-07-31T16:00:19.996450',
+        'id': 5715999101812736,
+        'layerId': 'elastic_maps_service.World Countries'
+      };
+    });
+
+    afterEach(function () {
+      if (window._skipTeardown) {
```

Comment:
Where does this variable come from? I can't find any reference to it in the repo.

Reply:
i used this for debugging, I need to remove it.